### PR TITLE
Refactoring usages of renderer/utils/createStorage

### PR DESCRIPTION
**Pull request by ixrock**
Follow-up for #2279 

- refactored usages of `renderer/utils/createStorage`
- refactored `Sidebar`, `SidebarItem` and `MainLayout`
- expanding sidebar by clicking arrow-icon (for consistency with compact mode where no sub-menus)

### CODE REVIEW

File: src/renderer/components/layout/sidebar-item.tsx [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+  icon?: React.ReactNode;
+  isHidden?: boolean;
+  isActive?: boolean;
+  subMenus?: TabLayoutRoute[];
+  onToggle?(id: string, meta: { props: SidebarItemProps, event: React.MouseEvent }): void;
+}
+
+@observer
+export class SidebarItem extends React.Component<SidebarItemProps> {
+  static displayName = "SidebarItem";
```
Comment (ixrock) [inline src/renderer/components/layout/sidebar-item.tsx:25]: Probably we should start using this static field in other places too to improve debugging react-components in `production`.

Comment (Nokel81) [reply src/renderer/components/layout/sidebar-item.tsx:25]: Should that be a PR of its own?

**Comment by ixrock** [comment]
@Nokel81 PTAL / re-review. 

I'm suggesting to merge this first since targeted to #2279 / **4.2.0** (not a patch release anymore) and continue reviewing in target-branch. Wdyt?

**Comment by Nokel81** [comment]
Sounds good, a single place to review is better.

File: src/renderer/components/layout/sidebar-item.tsx (hunk 1 above)
Comment (ixrock) [reply src/renderer/components/layout/sidebar-item.tsx:25]: Maybe, it's just an example..

**Pull request merged**